File: app/src/main/java/com/harsha/data/model/Illustration.java
package com.harsha.data.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Illustration implements Parcelable
{
    @SerializedName("imageUrl")
    @Expose
    private String imageUrl;

    @SerializedName("caption")
    @Expose
    private String caption;

    @SerializedName("description")
    @Expose
    private String description;

    protected Illustration(Parcel in) {
        imageUrl = in.readString();
        caption = in.readString();
        description = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(imageUrl);
        dest.writeString(caption);
        dest.writeString(description);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Illustration> CREATOR = new Creator<Illustration>() {
        @Override
        public Illustration createFromParcel(Parcel in) {
            return new Illustration(in);
        }

        @Override
        public Illustration[] newArray(int size) {
            return new Illustration[size];
        }
    };

    public String getImageUrl ()
    {
        return imageUrl;
    }

    public void setImageUrl (String imageUrl)
    {
        this.imageUrl = imageUrl;
    }

    public String getCaption ()
    {
        return caption;
    }

    public void setCaption (String caption)
    {
        this.caption = caption;
    }

    public String getDescription ()
    {
        return description;
    }

    public void setDescription (String description)
    {
        this.description = description;
    }

    @Override
    public String toString()
    {
        return "ClassPojo [imageUrl = "+imageUrl+", caption = "+caption+", description = "+description+"]";
    }
}

File: app/src/main/java/com/harsha/data/model/Illustration.kt
package com.harsha.data.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Illustration() :Parcelable{
    @SerializedName("imageUrl")
    @Expose
    private var imageUrl: String? = null

    @SerializedName("caption")
    @Expose
    private var caption: String? = null

    @SerializedName("description")
    @Expose
    private var description: String? = null

    constructor(parcel: Parcel) : this() {
        imageUrl = parcel.readString()
        caption = parcel.readString()
        description = parcel.readString()
    }

    fun getImageUrl(): String? {
        return imageUrl
    }

    fun setImageUrl(imageUrl: String?) {
        this.imageUrl = imageUrl
    }

    fun getCaption(): String? {
        return caption
    }

    fun setCaption(caption: String?) {
        this.caption = caption
    }

    fun getDescription(): String? {
        return description
    }

    fun setDescription(description: String?) {
        this.description = description
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(imageUrl)
        parcel.writeString(caption)
        parcel.writeString(description)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Illustration> {
        override fun createFromParcel(parcel: Parcel): Illustration {
            return Illustration(parcel)
        }

        override fun newArray(size: Int): Array<Illustration?> {
            return arrayOfNulls(size)
        }
    }
}